### PR TITLE
Unlock upgrade deploy script

### PR DESCRIPTION
**Pull request by clemsos**
# Description

Hardhat script to deploy new version of Unlock -- see v9 at #7348

# Issues

Refs  #7348

also help address #7436

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — APPROVED** [review]
LGTM :)

**Review by julien51 — CHANGES_REQUESTED** [review]
CCan we make this script agnostic and upgrade any of our contracts?

File: smart-contracts/scripts/unlock-upgrade.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  const chainId = await web3.eth.net.getId()
+  const networkName = process.env.RUN_MAINNET_FORK
+    ? 'mainnet'
+    : getNetworkName(chainId)
+
+  // eslint-disable-next-line no-console
+  console.log(`Deploying new implemntation on ${networkName}...`)
+
+  const [UDTInfo] =
+    OZ_SDK_EXPORT.networks[networkName].proxies['unlock-protocol/Unlock']
```
Comment (julien51) [inline smart-contracts/scripts/unlock-upgrade.js:16]: Do u thin we could make this a cli param?

Comment (clemsos) [reply smart-contracts/scripts/unlock-upgrade.js]: ok, much simpler :)

```
npx hardhat upgrade --contract contracts/Unlock.sol --network localhost
```

**Review by julien51 — COMMENTED** [review]
I think we're missing a file here :)

**Comment by clemsos** [comment]
Nope, the script was converted to a task, so it lives directly in the hardhat config now

**Comment by julien51** [comment]
Oh very cool!

**Review by julien51 — APPROVED** [review]

**Pull request merged**